pkg/controller/queuejobresources/genericresource/genericresource.go

```diff
@@ -69,31 +70,30 @@ func join(strs ...string) string {
 }
 
 func (gr *GenericResources) SyncQueueJob(aw *arbv1.AppWrapper, awr *arbv1.AppWrapperGenericResource) (podList []*v1.Pod, err error) {
-
 	startTime := time.Now()
 	defer func() {
-		glog.V(4).Infof("Finished syncing AppWrapper job resource %s (%v)", aw.Name, time.Now().Sub(startTime))
-		// glog.V(4).Infof("Finished syncing AppWrapper job resource %q (%v)", awobRes.Template, time.Now().Sub(startTime))
+		klog.V(4).Infof("Finished syncing AppWrapper job resource %s (%v)", aw.Name, time.Now().Sub(startTime))
+		// klog.V(4).Infof("Finished syncing AppWrapper job resource %q (%v)", awobRes.Template, time.Now().Sub(startTime))
 	}()
 
 	namespaced := true
 	//todo:DELETEME	dd := common.KubeClient.Discovery()
 	dd := gr.clients.Discovery()
 	apigroups, err := restmapper.GetAPIGroupResources(dd)
 	if err != nil {
-		glog.Fatal(err)
+		klog.Fatal(err)
 	}
 	ext := awr.GenericTemplate
 	restmapper := restmapper.NewDiscoveryRESTMapper(apigroups)
-	versions := &runtime.VersionedObjects{}
+	versions := &unstructured.Unstructured{}
```

Comment:
I see that later versions of K8s removed the `runtime.VersionsObjects{}`.  The good news is that the `Decode()` call below has code which should be able to handle not setting the versions (see `pkg/apis/meta/v1/unstructured/helpers.go` in the `apimachinery` library).  I'm also concerned about the warning provided in the`Unstructured{}` type:
```
// WARNING: This object has accessors for the v1 standard metadata. You *MUST NOT* use this
// type if you are dealing with objects that are not in the server meta v1 schema.
```
It also seems that the orig. code had the same logic not requiring the `versions` so I'm not quite sure why it was not used originally.  Can you remove the `versions` definition and replace it with a `nil` in the call below to test the proposed solution?